src/connector/src/source/filesystem/opendal_source/s3_source.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::marker::PhantomData;
+
+use anyhow::Context;
+use opendal::layers::{LoggingLayer, RetryLayer};
+use opendal::services::S3;
+use opendal::Operator;
+
+use super::opendal_enumerator::OpendalEnumerator;
+use super::OpenDalSourceProperties;
+// use crate::source::filesystem::opendal_source::get_prefix;
+use crate::source::filesystem::{s3::enumerator::get_prefix, S3Properties};
+
+impl<C: OpenDalSourceProperties> OpendalEnumerator<C>
+where
+    C: Sized + Send + Clone + PartialEq + 'static + Sync,
+{
+    /// create opendal s3 source.
+    pub fn new_s3_source(s3_properties: S3Properties) -> anyhow::Result<Self> {
+        // Create s3 builder.
+        let mut builder = S3::default();
+        builder.bucket(&s3_properties.bucket_name);
+        builder.region(&s3_properties.region_name);
+
+        if let Some(endpoint_url) = s3_properties.endpoint_url {
+            builder.endpoint(&endpoint_url);
+        }
+
+        if let Some(access) = s3_properties.access {
+            builder.access_key_id(&access);
+        }
+
+        if let Some(secret) = s3_properties.secret {
+            builder.secret_access_key(&secret);
+        }
```

Comment:
Yes.

Reply:
After checking the opendal documentation I found that it provides `role_arn()`, let me add it. 
BTW, we need to test it more on gcs because this interface has not been tested by opendal yet.